app/controllers/api/service_requests_controller.rb

```diff
@@ -45,16 +45,29 @@ def service_requests_search_conditions
       {:requester => User.current_user}
     end
 
-    def add_approver_resource(type, id, data)
+    def get_user(data)
       user_id = data['user_id'] || parse_id(data['user'], :users)
       raise 'Must specify a valid user_id or user' unless user_id
-      user = User.find(user_id)
+      User.find(user_id)
+    end
+
+    def add_approver_resource(type, id, data)
+      user = get_user(data)
       miq_approval = MiqApproval.create(:approver => user)
       resource_search(id, type, collection_class(:service_requests)).tap do |service_request|
         service_request.miq_approvals << miq_approval
       end
     rescue => err
       raise BadRequestError, "Cannot add approver - #{err}"
     end
+
+    def remove_approver_resource(type, id, data)
+      user = get_user(data)
+      resource_search(id, type, collection_class(:service_requests)).tap do |service_request|
+        service_request.miq_approvals.find_by(:approver_name => user.name).destroy
+      end
+    rescue => err
+      raise BadRequestError, "Cannot remove approver - #{err}"
```

Comment:
if the user is not an approver, the error message may not be nice, i.e. Cannot remove approver - undefined method `destroy' for nil:NilClass.  Please check that case and simply not do the destroy (i.e. should just be a success since the user is not an approver).  Thanks.